spec/controllers/api/documents_controller_spec.rb

```diff
@@ -75,13 +75,81 @@
       end
     end
 
+    context 'when searching by page type' do
+      let!(:english_article) do
+        create(:english_article, site: site)
+      end
+      let!(:review_article) do
+        create(
+          :page,
+          site: site,
+          label: 'review label',
+          layout: create(:layout, identifier: 'review')
+        )
+      end
+
+      before do
+        get '/api/en/documents', { document_type: ['review'] }, headers
+      end
+
+      it 'excludes other documents not within the page type' do
+        expect(documents.size).to be(1)
+      end
+
+      it 'returns documents only of the requested page type' do
+        expect(documents.first).to include(
+          'label' => 'review label',
+          'layout_identifier' => 'review'
+        )
+      end
+    end
+
+    context 'when searching by tag' do
+      let!(:page_with_tag) do
+        create(
+          :page_with_tag,
+          label: 'Page with tag',
+          site: site,
+          tag_name: 'pensions',
+        )
+      end
+      let!(:page_without_tag) do
+        create(:page, site: site)
+      end
+      let!(:page_with_a_different_tag) do
+        create(:page_with_tag, site: site, tag_name: 'sorry-not-a-pensions')
+      end
+
+      context 'when searching by a tag' do
```

Comment:
the nested context description is the same as its parent, please amend one or the other to make more readable

Reply:
♻️